test/e2e/utils/verification/scylladbcluster/verify.go

```diff
@@ -54,72 +55,92 @@ func Verify(ctx context.Context, sc *scyllav1alpha1.ScyllaDBCluster, rkcClusterM
 				condType: "Degraded",
 				status:   metav1.ConditionFalse,
 			},
+		}
 
-			// Controller conditions
-			{
-				condType: "RemoteScyllaDBDatacenterControllerProgressing",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteScyllaDBDatacenterControllerDegraded",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteNamespaceControllerProgressing",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteNamespaceControllerDegraded",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteServiceControllerProgressing",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteServiceControllerDegraded",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteEndpointSliceControllerProgressing",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteEndpointSliceControllerDegraded",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteEndpointsControllerProgressing",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteEndpointsControllerDegraded",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteRemoteOwnerControllerProgressing",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteRemoteOwnerControllerDegraded",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteConfigMapControllerProgressing",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteConfigMapControllerDegraded",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteSecretControllerProgressing",
-				status:   metav1.ConditionFalse,
-			},
-			{
-				condType: "RemoteSecretControllerDegraded",
-				status:   metav1.ConditionFalse,
-			},
+		for _, dc := range sc.Spec.Datacenters {
+			dcCondList := []condValue{
+				// Datacenter aggregated conditions
+				{
+					condType: fmt.Sprintf("Datacenter%sAvailable", dc.Name),
+					status:   metav1.ConditionTrue,
+				},
+				{
+					condType: fmt.Sprintf("Datacenter%sProgressing", dc.Name),
+					status:   metav1.ConditionFalse,
+				},
+				{
+					condType: fmt.Sprintf("Datacenter%sDegraded", dc.Name),
+					status:   metav1.ConditionFalse,
+				},
+
+				// Datacenter controller conditions
+				{
+					condType: fmt.Sprintf("RemoteScyllaDBDatacenterControllerDatacenter%sProgressing", dc.Name),
+					status:   metav1.ConditionFalse,
+				},
+				{
+					condType: fmt.Sprintf("RemoteScyllaDBDatacenterControllerDatacenter%sDegraded", dc.Name),
+					status:   metav1.ConditionFalse,
+				},
+				{
+					condType: fmt.Sprintf("RemoteNamespaceControllerDatacenter%sProgressing", dc.Name),
+					status:   metav1.ConditionFalse,
+				},
+				{
+					condType: fmt.Sprintf("RemoteNamespaceControllerDatacenter%sDegraded", dc.Name),
```

Comment:
These tests should probably not be reinventing the wheel of assembling condition names? (and instead use the primitives we are adding in this PR)

Reply:
In a general case I would agree, but I see this situation slightly differently: the status conditions in this PR are part of its public interface, and the helper function that represents the condition name based on its components can be used interchangeably with the verbatim string condition name.

With an assumption that the parametric condition naming (somewhat trivial) logic can be more than adequately tested on the unit level, I'm coming from the perspective that it should be perfectly adequate to pull it "outside" of the system under test.

The reason it's beneficial is that it helps focus the test on the logic that really matters, and prevents the "[change detector](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html)" behavior/anti-pattern.

Reply:
While I don't agree it's a change detector I applied your remark  to not pursue further discussion.

Reply:
after offline discussion i have applied the change, ptal